default.nix

```diff
@@ -128,6 +128,18 @@ let iosSupport = system == "x86_64-darwin";
           binutils-unwrapped = super.binutils-unwrapped.override {
             autoreconfHook = lib.optional self.stdenv.buildPlatform.isDarwin super.autoreconfHook269;
           };
+
+          haskell = super.haskell // {
+            compiler = super.haskell.compiler // super.lib.optionalAttrs (super.stdenv.isDarwin) {
+              ghc865 = super.haskell.compiler.ghc865.overrideAttrs (old: {
+                patches = (old.patches or []) ++ [ ./haskell-overlays/patches/ghc865/fix-big-sur.patch ];
```

Comment:
Can we put all the patches in the same place? some are defined here in the overlays and others are up around like 41.

Reply:
Generally I would like to have all compiler patches that we keep (in-repo) to be in a format like this
`$GIT_REPO/haskell-overlays/patches/<ghcver>/<platform>/*.patch`
or alternatively replace
`$GIT_REPO/haskell-overlays/patches` to just `$GIT_REPO/patches`
keeps away a ton of ugly directory tree layouts

Reply:
I didn't mean the patch files themselves, I meant the list of patches being applied to our compilers.

Reply:
https://github.com/reflex-frp/reflex-platform/blob/5aa890fdf2ddb1b21c55d349aa500213d9de8b44/default.nix#L41-L45

Reply:
New commit should make this align more with what we're currently doing